### PR TITLE
Converted RealmObjectTests to JUnit4

### PR DESCRIPTION
**Pull request by cmelchior**
@realm/java

### CODE REVIEW

File: realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java [hunk 1]
```diff
@@ -317,34 +363,42 @@ public void testEqualsStandAlone() {
...
+    public void hashCode_cyclicObject() {
+        // Don't use the configFactory as we need absolute control over the path to be able to calculate the hashCode
+        realm.close();
+        RealmConfiguration realmConfig = new RealmConfiguration.Builder(InstrumentationRegistry.getTargetContext()).build();
+        Realm.deleteRealm(realmConfig);
+        realm = Realm.getInstance(realmConfig);
 
+        realm.beginTransaction();
+        CyclicType foo = createCyclicData();
+        realm.commitTransaction();
```
Comment (beeender) [inline realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java]: Just close the Realm and delete it at then end of test.

File: realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java [hunk 2]
```diff
@@ -633,12 +691,13 @@ public void testSetClosedObjectToLinkLists() {
...
-            testRealm.cancelTransaction();
+            realm.cancelTransaction();
         }
     }
 
-    public void testSetObjectFromAnotherRealmToLinkLists() {
-        RealmConfiguration config = new RealmConfiguration.Builder(getContext()).name("another.realm").build();
+    @Test
+    public void setter_list_withObjectFromAnotherRealm() {
+        RealmConfiguration config = configFactory.createConfiguration("another.realm");
```
Comment (beeender) [inline realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java]: below `Realm.deleteRealm(config);` is redundant, just remove it.

File: realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java [hunk 3]
```diff
@@ -715,29 +775,32 @@ public void run() {
...
+    public void isValid_standaloneObject() {
         AllTypes allTypes = new AllTypes();
         assertFalse(allTypes.isValid());
     }
 
-    public void testIsValidClosedRealm() {
-        RealmConfiguration otherConfig = new RealmConfiguration.Builder(getContext()).name("other-realm").build();
+    @Test
+    public void isValid_closedRealm() {
+        RealmConfiguration otherConfig = configFactory.createConfiguration("other-realm");
```
Comment (beeender) [inline realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java]: remove the `Realm.deleteRealm(config);`

**Comment by beeender** [comment]
Besides a few minor comments, :+1:

File: realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java [hunk 4]
```diff
@@ -996,112 +1063,119 @@ public void run() {
...
+    public void float_double_NaN() {
+        realm.beginTransaction();
+        AllTypes allTypes = realm.createObject(AllTypes.class);
         allTypes.setColumnFloat(Float.NaN);
         allTypes.setColumnDouble(Double.NaN);
-        testRealm.commitTransaction();
-        assertEquals(Float.NaN, testRealm.where(AllTypes.class).findFirst().getColumnFloat());
-        assertEquals(Double.NaN, testRealm.where(AllTypes.class).findFirst().getColumnDouble());
+        realm.commitTransaction();
+        assertEquals(Float.NaN, realm.where(AllTypes.class).findFirst().getColumnFloat(), 0F);
```
Comment (kneth) [inline realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java]: `0F` -> `0.0f`

File: realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java [hunk 5]
```diff
@@ -996,112 +1063,119 @@ public void run() {
...
+        assertEquals(Double.NaN, realm.where(AllTypes.class).findFirst().getColumnDouble(), 0F);
```
Comment (kneth) [inline realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java]: `0F` -> `0.0d`

**Comment by kneth** [comment]
:+1: except minor comments on float and double literals.

**Pull request merged**